pact4s/src/main/scala/org/broadinstitute/dsde/workbench/cromwell/consumer/CbasClient.scala

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.dsde.workbench.cromwell.consumer
+
+import cats.effect.Concurrent
+import cats.syntax.all._
+import com.typesafe.scalalogging.LazyLogging
+import cromwell.engine.workflow.lifecycle.finalization.CallbackMessage
+import io.circe.Encoder
+import org.http4s._
+import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
+import org.http4s.client.Client
+import org.http4s.headers.Authorization
+
+trait CbasClient[F[_]] extends LazyLogging {
+  def postWorkflowResults(authHeader: Authorization,
+                          callbackMessage: CallbackMessage): F[Boolean]
+}
+
+/*
+ This class represents the consumer (Cromwell) view of the CBAS provider that implements the following endpoints:
+  - POST /api/batch/v1/runs/results
+ */
+class CbasClientImpl[F[_]: Concurrent](client: Client[F], baseUrl: Uri) extends CbasClient[F] {
+  val apiVersion = "v1"
+
+  implicit val workflowCallbackMessageEncoder: Encoder[CallbackMessage] = Encoder.forProduct4(
```

Comment:
Does Pact need a Circe JSON encoder rather than a Spray one? Maybe it would be possible to replace the other Spray one with this Circe one?

Reply:
The test needs to get outputs in JSON form. I used same references as used in engine.

Reply:
Can it use the existing JSON encoder, though, rather than creating a new one? If we import it it may just work - but I might misunderstand, maybe there's something about Spray that doesn't work here?